ArcGISRuntimeSDKQt_CppSamples/EditData/EditFeaturesWithFeatureLinkedAnnotation/EditFeaturesWithFeatureLinkedAnnotation.cpp

```diff
@@ -0,0 +1,272 @@
+
+  emit mapViewChanged();
+}
+
+void EditFeaturesWithFeatureLinkedAnnotation::onGeodatabaseDoneLoading(Error error)
+{
+  if (!error.isEmpty())
+    return;
+
+  GeodatabaseFeatureTable* pointFeatureTable = m_geodatabase->geodatabaseFeatureTable("Loudoun_Address_Points_1");
+  GeodatabaseFeatureTable* parcelLinesFeatureTable = m_geodatabase->geodatabaseFeatureTable("ParcelLines_1");
+
+  GeodatabaseFeatureTable* addressPointsAnnotationTable = m_geodatabase->geodatabaseAnnotationTable("Loudoun_Address_PointsAnno_1");
+  GeodatabaseFeatureTable* parcelLinesAnnotationTable = m_geodatabase->geodatabaseAnnotationTable("ParcelLinesAnno_1");
+
+  // create feature layers from tables in the geodatabase
+  m_pointFeatureLayer = new FeatureLayer(pointFeatureTable, this);
+  m_parcelLinesFeatureLayer = new FeatureLayer(parcelLinesFeatureTable, this);
+
+  // create annotation layers from tables in the geodatabase
+  m_addressPointsAnnotationLayer = new AnnotationLayer(addressPointsAnnotationTable, this);
+  m_parcelLinesAnnotationLayer = new AnnotationLayer(parcelLinesAnnotationTable, this);
+
+  // add feature layers to map
+  m_map->operationalLayers()->append(m_pointFeatureLayer);
+  m_map->operationalLayers()->append(m_parcelLinesFeatureLayer);
+
+  // add annotation layers to map
+  m_map->operationalLayers()->append(m_addressPointsAnnotationLayer);
+  m_map->operationalLayers()->append(m_parcelLinesAnnotationLayer);
+}
+
+void EditFeaturesWithFeatureLinkedAnnotation::onMouseClicked(QMouseEvent mouseEvent)
+{
+  clearSelection();
+
+  if (m_selectedFeature)
+  {
+    // move feature to clicked locaiton if already selected
+    const Point clickedPoint = m_mapView->screenToLocation(mouseEvent.x(), mouseEvent.y());
+    moveFeature(clickedPoint);
+  }
+  else
+  {
+    // identify and select feature
+    m_mapView->identifyLayers(mouseEvent.x(), mouseEvent.y(), 10, false);
+  }
+}
+
+void EditFeaturesWithFeatureLinkedAnnotation::onIdentifyLayersCompleted(QUuid, QList<IdentifyLayerResult *> identifyResults)
+{
+  if (identifyResults.isEmpty())
+    return;
+
+  for (IdentifyLayerResult* identifyResult : identifyResults)
+  {
+    if (!identifyResult->geoElements().empty())
+    {
+      // only perform selections on Feature Layers
+      FeatureLayer* featureLayer = dynamic_cast<FeatureLayer*>(identifyResult->layerContent());
+
+      // if cast was succesful select features.
+      if (featureLayer)
+      {
+        m_selectedFeature = static_cast<Feature*>(identifyResult->geoElements().at(0));
+
+        // Prevent the feature from being deleted along with the identifyResult.
+        m_selectedFeature->setParent(this);
+
+        const GeometryType selectedFeatureGeomType = m_selectedFeature->geometry().geometryType();
+
+        if (selectedFeatureGeomType == GeometryType::Polyline)
+        {
+          const Geometry geom = m_selectedFeature->geometry();
+          const PolylineBuilder* polylineBuilder = new PolylineBuilder(geom, this);
```

Comment:
this looks like a space leak. new'ing up a PolylineBuilder every time this slot executes but the parent is `this`. Try some type of RAII technique. Looks like a simple solution might be to set m_selectedFeature as parent since that is explicitly deleted later.

Reply:
To avoid the space leaks, consider using a value object. You'd need to make sure it works for this use case, but it's the simplest pattern to use and avoids a heap allocation entirely.
```c++
const PolylineBuilder polylineBuilder(geom);
```
If that works with the scoping that you need, it would be worth trying. You get RAII for free with value objects.